stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/utils/Linkify.kt

```diff
@@ -0,0 +1,214 @@
+        if (m !is LinkMovementMethod) {
+            if (t.linksClickable) {
+                t.movementMethod = LinkMovementMethod.getInstance()
+            }
+        }
+    }
+
+    private fun applyLink(
+        url: String,
+        start: Int,
+        end: Int,
+        text: Spannable,
+    ) {
+        val urlSpanFactory = DEFAULT_SPAN_FACTORY
+        val span = urlSpanFactory(url)
+        text.setSpan(span, start, end, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
+    }
+
+    private fun makeUrl(
+        url: String,
+        prefixes: Array<String>,
+        matcher: Matcher,
+        filter: Linkify.TransformFilter?,
+    ): String {
+        var url = url
+        if (filter != null) {
+            url = filter.transformUrl(matcher, url)
+        }
+        var hasPrefix = false
+        for (i in prefixes.indices) {
+            if (url.regionMatches(0, prefixes[i], 0, prefixes[i].length, ignoreCase = true)) {
+                hasPrefix = true
+
+                // Fix capitalization if necessary
+                if (!url.regionMatches(0, prefixes[i], 0, prefixes[i].length, ignoreCase = false)) {
+                    url = prefixes[i] + url.substring(prefixes[i].length)
+                }
+                break
+            }
+        }
+        if (!hasPrefix && prefixes.isNotEmpty()) {
+            url = prefixes[0] + url
+        }
+        return url
+    }
+
+    private fun gatherLinks(
+        links: MutableList<LinkSpec>,
+        s: Spannable,
+        pattern: Pattern,
+        schemes: Array<String>,
+        matchFilter: Linkify.MatchFilter?,
+        transformFilter: Linkify.TransformFilter?,
+    ) {
+        val m = pattern.matcher(s)
+        while (m.find()) {
+            val start = m.start()
+            val end = m.end()
+            if (matchFilter == null || matchFilter.acceptMatch(s, start, end)) {
+                val url: String = makeUrl(m.group(0), schemes, m, transformFilter)
+                val spec = LinkSpec(url = url, start = start, end = end)
+                links.add(spec)
+            }
+        }
+    }
+
+    private fun pruneOverlaps(links: MutableList<LinkSpec>, text: Spannable) {
+        // Append spans added by Markwon to remove any overlap.
+        val urlSpans: Array<URLSpan> = text.getSpans(0, text.length, URLSpan::class.java)
+        urlSpans.forEach { span ->
+            val spec = LinkSpec(
+                markwonAddedSpan = span,
+                start = text.getSpanStart(span),
+                end = text.getSpanEnd(span)
+            )
+            links.add(spec)
+        }
+
+        links.sortWith(COMPARATOR)
+
+        var len = links.size
+        var i = 0
+        while (i < len - 1) {
+            val a: LinkSpec = links[i]
+            val b: LinkSpec = links[i + 1]
+            var remove = -1
+            if (a.start <= b.start && a.end > b.start) {
+                when {
+                    b.end <= a.end -> {
+                        remove = i + 1
```

Comment:
You can lift the attribution of `remove`. remove = when { [...] }

Reply:
Then I will have to add `else` branch to it and not sure if it would do any benefit  (or make any logical sense to even have it) here.